### ISSUE TITLE
Will there be support for VS 2022 preview?

### ISSUE THREAD

**Issue by serp777** [github/VisualStudio#2634]
**Is your feature request related to a problem? Please describe.**
GIthub extension for visual studio not available for vs 2022 preview

**Describe the solution you'd like**
Compile the github extension for visual studio 2022

**Describe alternatives you've considered**
N/A

**Comment by serp777** [comment]
Bump

**Comment by Feerip** [comment]
bump

**Comment by 3nth** [comment]
It appears there will not be. Or at least not fully. 2022 has most of the features built in. https://github.com/github/VisualStudio/pull/2642

The feature missing for me is the PR Review functionality. Will either keep 2019 around for that or check out this VS Code extension, but either way will miss  having that in 2022.
https://marketplace.visualstudio.com/items?itemName=GitHub.vscode-pull-request-github

**Comment by serp777** [comment]
> It appears there will not be. Or at least not fully. 2022 has most of the features built in. #2642
> 
> The feature missing for me is the PR Review functionality. Will either keep 2019 around for that or check out this VS Code extension, but either way will miss having that in 2022. https://marketplace.visualstudio.com/items?itemName=GitHub.vscode-pull-request-github

Its missing many BIG features. Like for example, connecting to github and cloning repositories. It does not have the github enterprise sign in at all. And like you said no PR support. Built in features are pretty weak actually. I want to be able to use visual studio as my github tool without needing any other tool and right now with vs 2022 that's not possible. I'm forced to either use command line or use github desktop or something. Its certainly not a deal breaker but would definitely like to see it.

**Comment by janecosham** [comment]
Cloneing and otherwise using Github Enterprise is working for me in VS2022. I haven't used it a lot, but the only thing I have found missing is the PR review functionality which I use a lot in VS2019.

**Comment by janecosham** [comment]
I cloned directly from the Git | Clone menu by putting in the url to the enterprise repo. I didn't need to enter credentials as I have other local clones I had already opened.

**Comment by FSDKO** [comment]
I also miss the PR Review in VS 2022. That is essential for my work.
Managing Pull Requests was the essential feature of that extension.

On https://devblogs.microsoft.com/visualstudio/where-are-my-favorite-extensions-in-visual-studio-2022/ you can find this statement:
![image](https://user-images.githubusercontent.com/26324364/142165485-687cc948-bbb6-441a-b0d8-26ddfb6f23c2.png)
And if you follow the link, you find nothing about Pull Requests.

**This statement is a lie!!**

All I can do with VS 2022 ist open the Pull Requests in browser. (Menu Git -> GitHub -> View Pull Requests)
Not very helpfull.

Now I go back to VS 2019 with that extension.

**Comment by serp777** [comment]
> I cloned directly from the Git | Clone menu by putting in the url to the enterprise repo. I didn't need to enter credentials as I have other local clones I had already opened.

Yeah that stinks. I don't want to have to dig through the github web portal. I like it when it brings up my entire github enterprise org in a modal and I can search and filter right there an vs for the repository I have permission for. Then I hit clone and bam its done.

**Comment by bchavez** [comment]
Same here. Would be nice to be able to check out a PR directly from VS like 2019. Most def **Visual Studio 2022** "built-in GH integration" is NOT at feature parity with this VS 2019 GH extension:

![devenv_4430](https://user-images.githubusercontent.com/478118/142769982-fcda8c0a-9482-4830-a3cf-19262e8bc25c.png)

-----

And in 2022, "View Pull Requests" opens a browser :disappointed: :
![devenv_4431](https://user-images.githubusercontent.com/478118/142769983-6c3e5db7-3c4e-4377-b757-81cef0fc4f80.png)
n.

**Comment by molnard** [comment]
bump

**Comment by AraHaan** [comment]
As much as I like the built in features, the ones included in VS2022 does not include an ``Commit with signoff (--signoff)`` option when committing and I have requirements on my codebases for signoffs on all of their commits. I even enforce it in pull requests too as part of CI.

**Comment by nahk-ivanov** [comment]
💯

To be honest, I don't see vanilla VS 2022 git features being any different from those in vanilla VS 2019. Saying

> Visual Studio 2022 now includes the functionality from this extension out of the box.

..is clearly a fake news.

And lets be honest - GitHub Web UI is not really designed to be used by human beings.

The matters made worse by the fact that .NET 6 is only supported by VS 2022, so you are either stuck with VS 2019 and .NET Core 3.1 (because .NET 5 is not LTS, its support ends sooner than Core 3.1), or VS 2022 and .NET 6 with no proper dev experience. One step forward and two steps back, IDE slowly becomes a beefed up text editor, it seems like..?

¯\_(ツ)_/¯

**Comment by AraHaan** [comment]
Honestly Rider is better than VS at this point, easier extensibility, easier config, cross platform, but not free.

**Comment by soroshsabz** [comment]
ITNOA

PR review is very important and useful feature. So I think if this plugin merged into VS 2022, so please add all features in this. I create this feature request in [Microsoft Developer Community](https://developercommunity.visualstudio.com/t/Bring-all-GitHub-extension-features-to-V/10074172), Please vote up to push Microsoft.

**Comment by FSDKO** [comment]
The feature request in [Microsoft Developer Community](https://developercommunity.visualstudio.com/t/Bring-all-GitHub-extension-features-to-V/10074172) posted by @soroshsabz was closed because of duplicate.
This had 11 votes.

The duplicate is [here](https://developercommunity.visualstudio.com/t/create-and-edit-GitHub-issues-and-Pull-R/10060891). This has only 7 votes yet. Please vote it.

**Comment by MichaConrad** [comment]
Actually.. :grin: @FSDKO, that issue with 7 votes is now also closed in favor [of this](https://developercommunity.visualstudio.com/t/cant-create-pull-request-in-visual-studio-with-new/1017696) with 150+ votes.

**Comment by soroshsabz** [comment]
My feature request is mark duplicate with https://developercommunity.visualstudio.com/t/VS2022-171-Git-Pull-Request-is-gone/1576559 but I think the original duplicate is @MichaCo says :)

**Comment by houghj16** [comment]
Hello all, 

Thanks for your engagement on this topic! We're currently investigating how best to bring this functionality to you, and I was hoping to gather your any of your additional thoughts on Pull Requests in VS with the following survey - https://www.research.net/r/VS2022PR. Please leave your contact info, if you'd like to participate in interviews or user studies as we develop this feature.

Best,
Jessie Houghton, PM for Version Control in VS

**Comment by frankiea** [comment]
@houghj16 Hi Jessie - any update on this survey and resolution? I'm still holding on to VS2019 because of this one crucial feature I use daily in my workflow.

**Comment by houghj16** [comment]
@frankiea, Thanks for your comment. We've placed the following - https://developercommunity.visualstudio.com/t/cant-create-pull-request-in-visual-studio-with-new/1017696 on our public roadmap, so you can track the progress there!
